src/edu/csus/ecs/pc2/core/model/RunFiles.java

```diff
@@ -178,16 +182,88 @@ public String toString() {
                 }
 
             }
-            
-        } 
+
+        }
         //close OtherFiles list
-        retStr += "]";  
-        
+        retStr += "]";
+
         //close retStr object
         retStr += "]";
-        
+
         return retStr ;
-        
+
     }
 
+    /**
+     * createCatalogJSON - Generate detailed JSON information about the files submitted and write it to the
+     * supplied BufferedWriter.  Similar to toString above, but supplies more details.
+     *
+     * @param catlog Where to send the detailed JSON about the submission files
+     */
+    public void createCatalogJSON(BufferedWriter catlog, Log log) {
+        try {
+            boolean needComma = false;
+            catlog.write("[");
+            if(mainFile != null) {
+                catlog.newLine();
+                catlog.write(" {\"mainfile\":");
+                catlog.write(serializedFileToJSON(mainFile));
+                catlog.write(" }");
+                needComma = true;
+            }
+            if(otherFiles != null && otherFiles.length > 0) {
+                boolean needComma2 = false;
+                if(needComma) {
+                    catlog.write(',');
+                }
+                catlog.newLine();
+                catlog.write(" {\"otherfiles\": {");
+                for(SerializedFile f : otherFiles) {
+                    if(needComma) {
+                        catlog.write(',');
+                        catlog.newLine();
+                    }
+                    catlog.write("  ");
+                    catlog.write(serializedFileToJSON(f));
+                }
+                catlog.newLine();
+                catlog.write(" }");
+                // in case we add more stuff to the array later
+                needComma = true;
+            }
+            catlog.write(']');
+            catlog.newLine();
+        } catch (IOException e) {
+            log.log(Log.WARNING, "Exception writing RunFiles catalog file for run " + getRunId().toString(), e);
+        }
+    }
+
+    /**
+     * serializedFileToJSON - pick out the detailed info we want from the SerializedFile
+     *     and create a JSON string.  Ex.
+     *     { "absolutePath":"C:/home/icpc/pc2/current/config/atotientquotient/submissions/time_limit_exceeded/evouga-brute.cpp",
+     *       "name":"evouga-brute.cpp",
+     *       "fileType":2,
+     *       "externalFile":false
+     *     }
+     * @param f the serialized file
+     * @return JSON string representation of the detailed info
+     */
+    private String serializedFileToJSON(SerializedFile f) {
```

Comment:
The name of this method seems a bit misleading; it implies that it's creating a JSON string for the serialized file (whereas what it's actually doing is creating a JSON string for the "detailed info" of the serialized file).